### PR TITLE
Add Net.WalkPrefix API

### PR DESCRIPTION
**Pull request by fujita**
For interating routes that have a given prefix.

Signed-off-by: FUJITA Tomonori <fujita.tomonori@gmail.com>

### CODE REVIEW

**Comment by k-sone** [comment]
Thank you for sending a pull request.

I wrote a sample code and ran it.
```go
package main
 
import (
        "fmt"
        "net"
 
        "github.com/k-sone/critbitgo"
)

func testWalkPrefix(trie *critbitgo.Net, n string) {
        f := func(n *net.IPNet, _ interface{}) bool {
                fmt.Println(n)
                return true
        }

        fmt.Printf("Case of %s\n", n)
        _, s, _ := net.ParseCIDR(n)
        trie.WalkPrefix(s, f)
}

func main() {
        trie := critbitgo.NewNet()
        trie.AddCIDR("192.168.0.0/24", nil)
        trie.AddCIDR("192.168.1.0/24", nil)
        trie.AddCIDR("192.168.2.0/24", nil)

        testWalkPrefix(trie, "192.168.0.0/24")
        testWalkPrefix(trie, "192.168.0.0/23")
}
```

And I got the output.
```
Case of 192.168.0.0/24
192.168.0.0/24
Case of 192.168.0.0/23
192.168.0.0/24
192.168.1.0/24
192.168.2.0/24
```

I think the last line is unnecessary, Is this your expected results?

**Comment by fujita** [comment]
Oops, I messed up. Fixed the handling of prefix length, non multiple of 8.
Thanks for the quick response!

File: net.go [hunk 1]
```diff
@@ -203,6 +203,33 @@ func (n *Net) Walk(r *net.IPNet, handle func(*net.IPNet, interface{}) bool) {
...
+			div = int(mask >> 3)
+			if mod := uint(mask & 0x07); mod != 0 {
+				bit = 8 - mod
+			}
+		}
+	}
+	wrapper := func(key []byte, value interface{}) bool {
+		if bit != 0 {
+			if prefix[div]>>bit != key[div]>>bit {
+				return true
```
Comment (k-sone) [inline net.go]:
```diff
-				return true
+				return false
```
If true, I think will unnecessary repetition occurs in `Allprefixed`.

Comment (fujita) [reply net.go]: Ah, fixed. Thanks!

**Comment by k-sone** [comment]
Thanks!

**Pull request merged**